src/plugins/chessevent/engine/action_selector.py

```diff
@@ -34,6 +41,144 @@
 class ActionSelector(metaclass=Singleton):
     """The CLI interface for ChessEvent."""
 
+    @classmethod
+    def add_chessevent_player(
+            cls, database: PapiDatabase, player_papi_id: int, player: ChessEventPlayer, check_in_started: bool
+    ):
+        """Creates a player in the database from the given ChessEvent player.
+        If the player is not checked in when `check_in_started` is True,
+        removes the player from play for subsequent rounds which are not
+        specifically not-played rounds."""
+        data: dict[str, str | int | float | None] = {
+            'Ref': player_papi_id,
+            'RefFFE': player.ffe_id,
+            'NrFFE': player.ffe_license_number if player.ffe_license_number else None,
+            'Nom': player.last_name,
+            'Prenom': player.first_name,
+            'Sexe': player.gender.to_papi_value,
+            'NeLe': database.timestamp_to_papi_date(player.birth),
+            'Cat': player.category.to_papi_value,
+            'AffType': player.ffe_license.to_papi_value,
+            'Elo': player.standard_rating,
+            'Rapide': player.rapid_rating,
+            'Blitz': player.blitz_rating,
+            'Federation': player.federation,
+            'ClubRef': player.ffe_club_id,
+            'Club': player.ffe_club,
+            'Ligue': player.ffe_league,
+            'Fide': player.standard_rating_type.to_papi_value,
+            'RapideFide': player.rapide_rating_type.to_papi_value,
+            'BlitzFide': player.blitz_rating_type.to_papi_value,
+            'FideCode': player.fide_id if player.fide_id else None,
+            'FideTitre': player.title.to_papi_value,
+            'Pointe': check_in_started and player.check_in,
+            'InscriptionRegle': player.paid,
+            'InscriptionDu': player.fee,
+            'Tel': player.phone,
+            'EMail': player.email,
+            'Fixe': player.board,
+            'Flotteur': 'X' * 24,
+            'Pts': 0,
+            'PtA': 0,
+        }
+        for round_ in range(1, 25):
+            data[f'Rd{round_:0>2}Adv'] = None
+            if round_ not in player.skipped_rounds:
+                data[f'Rd{round_:0>2}Res'] = Result.NO_RESULT.to_papi_value
+                if player.check_in or not check_in_started:
+                    data[f'Rd{round_:0>2}Cl'] = 'R'
+                else:
+                    data[f'Rd{round_:0>2}Cl'] = 'F'
+            else:
+                data[f'Rd{round_:0>2}Cl'] = 'F'
+                match player.skipped_rounds[round_]:
+                    case 0.0:
+                        data[f'Rd{round_:0>2}Res'] = Result.NO_RESULT.to_papi_value
+                    case 0.5:
+                        data[f'Rd{round_:0>2}Res'] = Result.HALF_POINT_BYE.to_papi_value
+                    case _:
+                        raise ValueError
+        query: str = f'INSERT INTO `joueur`({", ".join(data.keys())}) VALUES ({", ".join(["?"] * len(data))})'
```

Comment:
I'd like to see `:AAAA` parameters if possible (i.e. if the papi database supports it)

Reply:
That code has just been moved from another file.  Please make a change on a new PR if you wish to